### ISSUE TITLE
book中数字识别的例子，cost最小的不是acc最高的epoch

### ISSUE THREAD

**Issue by cissoidx** [PaddlePaddle/Paddle#26587]
直接运行官方示例，多次出现标题中的情况。

Paddle version: 1.8.4
Paddle With CUDA: True

OS: debian buster/sid
Python version: 3.6.8

CUDA version: 10.1.243
cuDNN version: 7.6.5
Nvidia driver version: 440.64

![image](https://user-images.githubusercontent.com/17111388/90999734-f51a9980-e5f9-11ea-9a2a-64d87dfec849.png)

为使您的问题得到快速解决，在建立Issues前，请您先通过如下方式搜索是否有相似问题:【搜索issue关键字】【使用labels筛选】【官方文档】

如果您没有查询到相似问题，为快速解决您的提问，建立issue时请提供如下细节信息：
- 标题：简洁、精准概括您的问题，例如“Insufficient Memory xxx" ”
- 版本、环境信息：
    1）PaddlePaddle版本：请提供您的PaddlePaddle版本号，例如1.1或CommitID
    2）CPU：预测若用CPU，请提供CPU型号，MKL/OpenBlas/MKLDNN/等数学库使用情况
    3）GPU：预测若用GPU，请提供GPU型号、CUDA和CUDNN版本号
    4）系统环境：请您描述系统类型、版本，例如Mac OS 10.14，Python版本
 注：您可以通过执行[summary_env.py](https://github.com/PaddlePaddle/Paddle/blob/develop/tools/summary_env.py)获取以上信息。
- 训练信息
    1）单机/多机，单卡/多卡
    2）显存信息
    3）Operator信息
- 复现信息：如为报错，请给出复现环境、复现步骤
- 问题描述：请详细描述您的问题，同步贴出报错信息、日志、可复现的代码片段

Thank you for contributing to PaddlePaddle.
Before submitting the issue, you could search issue in the github in case that there was a similar issue submitted or resolved before.
If there is no solution,please make sure that this is a training issue including the following details:
**System information**
-PaddlePaddle version （eg.1.1）or CommitID
-CPU: including CPUMKL/OpenBlas/MKLDNN version
-GPU: including CUDA/CUDNN version
-OS Platform (eg.Mac OS 10.14)
-Other imformation: Distriuted training/informantion of operator/
Graphics card storage
Note: You can get most of the information by running [summary_env.py](https://github.com/PaddlePaddle/Paddle/blob/develop/tools/summary_env.py). 
**To Reproduce**
Steps to reproduce the behavior
**Describe your current behavior**
**Code to reproduce the issue**
**Other info / logs**

**Comment by willthefrog** [comment]
loss function和validation metrics存在gap是非常正常的现象，overfitting就是最典型的例子。

**Comment by cissoidx** [comment]
哦，对的。被输出的文字有点误导了。不过，这个跟overfitting关系不大吧。
